File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        stack = [[root,float('-inf')]]
        count = 0
        while stack:
            node,value = stack.pop()
            maxx = value
            if node.val >= value:
                count += 1
                maxx = node.val
            if node.left:
                stack.append([node.left,maxx])
            if node.right:
                stack.append([node.right,maxx])
        return(count)


        

File: Data Structures & Algorithms/validate-parentheses/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        let stack = [];
        let map = {
            '}': '{',
            ']': '[',
            ')': '(',
        };
        for(let char of s){
           // let isBracket = char in map;
            if(!(char in map)){
             stack.push(char);
             }else{
                if(stack[stack.length -1] == map[char] && stack.length >= 1){
                    stack.pop()
                }else{
                    return false
                }
             }
         }
        console.log(stack)
        return stack.length === 0;
      }
    }
